app/controllers/application_controller.rb

```diff
@@ -28,7 +28,7 @@ def with_current_request
     ManageIQ::API::Common::Request.with_request(request) do |current|
       begin
         if Tenant.tenancy_enabled?
-          tenant = Tenant.find_by(:external_tenant => current.user.tenant)
+          tenant = Tenant.find_or_create_by(:external_tenant => current.user.tenant) if current.user.tenant
```

Comment:
I believe find_or_create_by is a multi-threaded race condition...need to deal with that or we will have duplicate tenants.

Reply:
There's a unique constraint on tenants.external_tenant